src/badge/examples/SCRHoldingBadge.sol

```diff
@@ -0,0 +1,150 @@
+
+function decodePayloadData(bytes memory data) pure returns (uint256) {
+    return abi.decode(data, (uint256));
+}
+
+/// @title SCRHoldingBadge
+/// @notice A badge that represents user's SCR holding amount.
+contract SCRHoldingBadge is ScrollBadgeCustomPayload, ScrollBadgeDefaultURI, Ownable, IScrollSelfAttestationBadge {
+    uint256 private constant LEVEL_ONE_SCR_AMOUNT = 1 ether;
+    uint256 private constant LEVEL_TWO_SCR_AMOUNT = 10 ether;
+    uint256 private constant LEVEL_THREE_SCR_AMOUNT = 100 ether;
+    uint256 private constant LEVEL_FOUR_SCR_AMOUNT = 1000 ether;
+    uint256 private constant LEVEL_FIVE_SCR_AMOUNT = 10_000 ether;
+    uint256 private constant LEVEL_SIX_SCR_AMOUNT = 100_000 ether;
+
+    /// @notice The address of SCR token.
+    address public immutable scr;
+
+    constructor(address resolver_, string memory baseTokenURI_, address scr_)
+        ScrollBadge(resolver_)
+        ScrollBadgeDefaultURI(baseTokenURI_)
+    {
+        scr = scr_;
+    }
+
+    /// @notice Update the base token URI.
+    /// @param baseTokenURI_ The new base token URI.
+    function updateBaseTokenURI(string memory baseTokenURI_) external onlyOwner {
+        defaultBadgeURI = baseTokenURI_;
+    }
+
+    /// @inheritdoc ScrollBadge
+    function onIssueBadge(Attestation calldata)
+        internal
+        virtual
+        override (ScrollBadge, ScrollBadgeCustomPayload)
+        returns (bool)
+    {
+        return false;
+    }
+
+    /// @inheritdoc ScrollBadge
+    function onRevokeBadge(Attestation calldata)
+        internal
+        virtual
+        override (ScrollBadge, ScrollBadgeCustomPayload)
+        returns (bool)
+    {
+        return false;
+    }
+
+    /// @inheritdoc ScrollBadge
+    function badgeTokenURI(bytes32 uid)
+        public
+        view
+        override (IScrollBadge, ScrollBadge, ScrollBadgeDefaultURI)
+        returns (string memory)
+    {
+        return ScrollBadgeDefaultURI.badgeTokenURI(uid);
+    }
+
+    /// @inheritdoc IScrollBadge
+    function hasBadge(address user) public view virtual override (IScrollBadge, ScrollBadge) returns (bool) {
+        uint256 balance = IERC20(scr).balanceOf(user);
+        return balance >= LEVEL_ONE_SCR_AMOUNT;
+    }
+
+    /// @inheritdoc ScrollBadgeDefaultURI
+    function getBadgeTokenURI(bytes32 uid) internal view override returns (string memory) {
+        Attestation memory attestation = getAndValidateBadge(uid);
+        bytes memory payload = getPayload(attestation);
+        uint256 level = decodePayloadData(payload);
+
+        return string(abi.encodePacked(defaultBadgeURI, Strings.toString(level), ".json"));
+    }
+
+    /// @inheritdoc ScrollBadgeCustomPayload
+    function getSchema() public pure override returns (string memory) {
+        return SCR_HOLDING_BADGE_SCHEMA;
+    }
+
+    /// @inheritdoc IScrollSelfAttestationBadge
+    function getBadgeId() external pure returns (uint256) {
+        return 0;
```

Comment:
What is this used for?

Reply:
This is just an unique identifier for self attested badged. No usage for now.